app/src/ereferrals/java/org/eyeseetea/malariacare/fragments/AVFragment.java

```diff
@@ -119,15 +120,15 @@ private void initializeChangeModeButtons() {
         gridMode.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                stylePreferencesRepository.saveMediaListMode(Settings.MediaListMode.GRID);
-                mPresenter.onClickChangeMode(Settings.MediaListMode.GRID);
+                settingsDataSource.saveMediaListMode(MediaListMode.GRID);
+                mPresenter.onClickChangeMode(MediaListMode.GRID);
             }
         });
         listMode.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                stylePreferencesRepository.saveMediaListMode(Settings.MediaListMode.LIST);
-                mPresenter.onClickChangeMode(Settings.MediaListMode.LIST);
+                settingsDataSource.saveMediaListMode(MediaListMode.LIST);
```

Comment:
a Fragment should not use the settingsDataSource, It's the presenter who should invoke request to save data in data layer but through of a use case, please create also SaveSettingsUseCase if not exists